modules/carto/src/layers/point-label-layer.ts

```diff
@@ -0,0 +1,266 @@
+import {
+  Accessor,
+  Color,
+  CompositeLayer,
+  CompositeLayerProps,
+  DefaultProps,
+  Layer,
+  LayersList
+} from '@deck.gl/core';
+import {
+  TextLayer,
+  TextLayerProps,
+  _TextBackgroundLayer as TextBackgroundLayer
+} from '@deck.gl/layers';
+
+const [LEFT, TOP, RIGHT, BOTTOM] = [0, 1, 2, 3];
+
+class EnhancedTextBackgroundLayer extends TextBackgroundLayer {
+  static layerName = 'EnhancedTextBackgroundLayer';
+
+  getShaders() {
+    const shaders = super.getShaders();
+    let vs = shaders.vs;
+
+    // Modify shader so that the padding is offset by the pixel offset to ensure the padding
+    // always captures the anchor point. As padding is uniform we cannot pass it a per-label value
+    vs = vs.replaceAll('padding.', '_padding.');
+    vs = vs.replace(
+      'void main(void) {',
+      'void main(void) {\n  vec4 _padding = padding + instancePixelOffsets.xyxy * vec4(1.0, 1.0, -1.0, -1.0);'
+    );
+
+    return {...shaders, vs};
+  }
+}
+
+// TextLayer which includes modified text-background-layer-vertex shader and only renders the
+// primary background layer in the collision pass
+class EnhancedTextLayer extends TextLayer {
+  static layerName = 'EnhancedTextLayer';
+
+  filterSubLayer({layer, renderPass}) {
+    const background = layer.id.includes('primary-background');
+    if (renderPass === 'collision') {
+      return background; // Only draw primary background layer in collision pass
+    }
+
+    return !background; // Do not draw background layer in other passes
+  }
+}
```

Comment:
It's a fairly specific use-case and `TextLayer` already has many options. Also it would introduce a new dependency of deck.gl/extensions in deck.gl/layers

Reply:
Makes sense to me. In general, all the stuff in the CARTO module could be promoted to a generic package if we see value for users.